crates/fullnode/src/runner.rs

```diff
@@ -186,467 +195,625 @@ where
-                    .extract_relevant_blobs(&filtered_block)
-                    .into_iter()
-                    .for_each(|mut tx| {
-                        let data = DaData::try_from_slice(tx.full_data());
-                        // Check for commitment
-                        if tx.sender().as_ref() == self.sequencer_da_pub_key.as_slice() {
-                            if let Ok(DaData::SequencerCommitment(seq_com)) = data {
-                                sequencer_commitments.push(seq_com);
-                            } else {
-                                tracing::warn!(
-                                    "Found broken DA data in block 0x{}: {:?}",
-                                    hex::encode(filtered_block.hash()),
-                                    data
-                                );
-                            }
-                        }
-                        let data = DaData::try_from_slice(tx.full_data());
-                        // Check for proof
-                        if tx.sender().as_ref() == self.prover_da_pub_key.as_slice() {
-                            if let Ok(DaData::ZKProof(proof)) = data {
-                                zk_proofs.push(proof);
-                            } else {
-                                tracing::warn!(
-                                    "Found broken DA data in block 0x{}: {:?}",
-                                    hex::encode(filtered_block.hash()),
-                                    data
-                                );
+                    .put_soft_confirmation_status(SlotNumber(i), SoftConfirmationStatus::Proven)?;
+            }
+        }
+        // store in ledger db
+        self.ledger_db.update_verified_proof_data(
+            l1_block.header().height(),
+            proof.clone(),
+            stored_state_transition,
+        )?;
+        Ok(())
+    }
+
+    async fn process_sequencer_commitment(
+        &self,
+        l1_block: Da::FilteredBlock,
+        sequencer_commitment: SequencerCommitment,
+    ) -> Result<(), SyncError> {
+        let start_l1_height =
+            get_da_block_by_hash(&self.da_service, sequencer_commitment.l1_start_block_hash)
+                .await?
+                .header()
+                .height();
+
+        let end_l1_height =
+            get_da_block_by_hash(&self.da_service, sequencer_commitment.l1_end_block_hash)
+                .await?
+                .header()
+                .height();
+
+        let start_l2_height = match self
+            .ledger_db
+            .get_l2_range_by_l1_height(SlotNumber(start_l1_height))?
+        {
+            Some((start_l2_height, _)) => start_l2_height,
+            None => {
+                return Err(anyhow!(
+                        "Sequencer commitment verification: L1 L2 connection does not exist. L1 height = {}. Skipping commitment.",
+                        start_l1_height
+                ).into());
+            }
```

Comment:
this can also happen due to a missing L2 block, can retry later if this happens

Reply:
I have added those cases to `SyncError::MissingL2` so that they can be retried.